File: moviejava/app/src/main/java/com/example/moviejava/ui/favorite/FavoriteAdapter.java
package com.example.moviejava.ui.favorite;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.moviejava.data.local.entities.MovieEntity;
import com.example.moviejava.databinding.ItemFavoriteBinding;
import com.example.moviejava.extentions.OnClickItem;

import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;

public class FavoriteAdapter extends RecyclerView.Adapter<FavoriteAdapter.ViewHolder> {
    private List<MovieEntity> list= new ArrayList<>();
    public OnClickItem<MovieEntity> onClick,onDelete;

    @Inject
    public FavoriteAdapter() {
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(
                ItemFavoriteBinding.inflate(
                        LayoutInflater.from(parent.getContext()),
                        parent,
                        false));
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.onBind(list.get(position));
        holder.itemView.setOnClickListener(
                view -> onClick.onClick(list.get(position))
        );
        holder.binding.deleteFavorite.setOnClickListener( view -> onDelete.onClick(list.get(position)));
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        private final ItemFavoriteBinding binding;

        public ViewHolder(@NonNull ItemFavoriteBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        public void onBind(MovieEntity m) {
            binding.setM(m);
            binding.executePendingBindings();
        }
    }

    public void submitData(List<MovieEntity> list) {
        this.list = list;
        notifyDataSetChanged();
    }
}

File: movie/app/src/main/java/com/example/movie/ui/favorite/FavoriteAdapter.kt
package com.example.movie.ui.favorite

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.animation.Animation
import android.view.animation.ScaleAnimation
import androidx.recyclerview.widget.RecyclerView
import com.example.movie.data.local.entities.MovieEntity
import com.example.movie.databinding.ItemFavoriteBinding
import javax.inject.Inject


class FavoriteAdapter @Inject constructor() : RecyclerView.Adapter<FavoriteAdapter.ViewHolder>() {
    private var companies = mutableListOf<MovieEntity>()
    var onClickItem: ((MovieEntity) -> Unit)? = null
    var onDelete: ((MovieEntity) -> Unit)? = null

    @SuppressLint("NotifyDataSetChanged")
    fun submitData(data: List<MovieEntity>) {
        companies.clear()
        companies.addAll(data)
        notifyDataSetChanged()
    }

    inner class ViewHolder(val binding: ItemFavoriteBinding) :
        RecyclerView.ViewHolder(binding.root) {
        fun onBind(m: MovieEntity) {
            binding.m = m
            binding.executePendingBindings()
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(
            ItemFavoriteBinding.inflate(
                LayoutInflater.from(parent.context),
                parent,
                false
            )
        )
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        companies[position].let {
            holder.onBind(it)
            holder.itemView.setOnClickListener { _ ->
                onClickItem?.invoke(it)
            }
            holder.binding.deleteFavorite.setOnClickListener { _ ->
                onDelete?.invoke(it)
            }
            //setScaleAnimation(holder.itemView)
        }
    }

    override fun getItemCount(): Int = companies.size
    private fun setScaleAnimation(view: View) {
        val anim = ScaleAnimation(
            0.0f,
            1.0f,
            0.0f,
            1.0f,
            Animation.RELATIVE_TO_SELF,
            0.5f,
            Animation.RELATIVE_TO_SELF,
            0.5f
        )
        anim.duration = 1000
        view.startAnimation(anim)
    }
}
